stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+                                 | (_Can_rep_day ? _F_doy : 0)};
+
+            const auto _Used{_Used_fields()};
+            const auto _Time_out_of_range{
+                _For_time_point
+                && (_Out_of_range(_Second, 0, 60) || _Out_of_range(_Minute, 0, 59) || _Out_of_range(_Hour_24, 0, 23))};
+
+            if (_Time_out_of_range || !_Test_bits(_Used, _Required, _Optional)) {
+                return false;
+            }
+
+            _Result = _DurationType::zero();
+            if constexpr (_Can_rep_sec) {
+                if (_Used & _F_sec) {
+                    if (_Subsecond) {
+                        using _CastedType = duration<int64_t, typename _DurationType::period>;
+                        const _SubsecondType _Sub{*_Subsecond};
+                        if (treat_as_floating_point_v<_DurationType> //
+                            || _CHRONO duration_cast<_CastedType>(_Sub) == _Sub) {
+                            _Result += _CHRONO duration_cast<_DurationType>(_Sub);
+                        } else {
+                            return false;
+                        }
+                    }
+
+                    if (_Second) {
+                        _Result += _CHRONO duration_cast<_DurationType>(seconds{*_Second});
+                    }
+                }
+            }
+
+            if constexpr (_Can_rep_min) {
+                if (_Used & _F_min) {
+                    _Result += _CHRONO duration_cast<_DurationType>(minutes{*_Minute});
+                }
+
+                if (_Utc_offset) {
+                    _Result -= _CHRONO duration_cast<_DurationType>(minutes{*_Utc_offset});
+                }
+            }
+
+            if constexpr (_Can_rep_hr) {
+                if (_Used & _F_hr) {
+                    _Result += _CHRONO duration_cast<_DurationType>(hours{*_Hour_24});
+                }
+            }
+
+            if constexpr (_Can_rep_day) {
+                if (_For_time_point) {
+                    year_month_day _Ymd;
+                    if (_Make_year_month_day(_Ymd, true)) {
+                        _Result += _CHRONO duration_cast<_DurationType>(static_cast<sys_days>(_Ymd).time_since_epoch());
+                    } else {
+                        return false;
+                    }
+                } else if (_Used & _F_doy) {
+                    _Result += _CHRONO duration_cast<_DurationType>(days{*_Day_of_year});
+                }
+            }
+
+            return true;
+        }
+
+        template <class _Rep, class _Period>
+        _NODISCARD bool _Make_duration(duration<_Rep, _Period>& _Duration_result) {
+            const bool _Consistent = _Calculate_hour24();
+            if (_Consistent) {
+                _Duration_result = duration<_Rep, _Period>::zero(); // TRANSITION: LWG-3536 & GH-1740
+            }
+
+            return _Consistent && _Apply_duration_fields(_Duration_result, false);
+        }

```

Comment:
I find it super icky that the `_Apply_duration_fields` call might not happen due to short cuircuiting. Maybe simplify to this
```suggestion
            if (_Calculate_hour24()) {
                _Duration_result = duration<_Rep, _Period>::zero(); // TRANSITION: LWG-3536 & GH-1740
                return _Apply_duration_fields(_Duration_result, false);
            } 
            return false;
        }
```